pandas/tests/frame/test_analytics.py

```diff
@@ -774,15 +776,19 @@ def _check_stat_op(self, name, alternative, frame=None, has_skipna=True,
             assert len(result)
 
         if has_skipna:
-            def skipna_wrapper(x):
-                nona = x.dropna()
-                if len(nona) == 0:
-                    return np.nan
-                return alternative(nona)
-
             def wrapper(x):
                 return alternative(x.values)
 
+            if skipna_alternative:
```

Comment:
This seemed like the easiest way of getting these tests to pass. `skipna_alternative` is only used in sum and prod, and is `nansum` or `nanprod`.

Reply:
this is ok